standard/variables.md

```diff
@@ -901,3 +901,181 @@ variable_reference
 ## 9.6 Atomicity of variable references
 
 Reads and writes of the following data types shall be atomic: `bool`, `char`, `byte`, `sbyte`, `short`, `ushort`, `uint`, `int`, `float`, and reference types. In addition, reads and writes of enum types with an underlying type in the previous list shall also be atomic. Reads and writes of other types, including `long`, `ulong`, `double`, and `decimal`, as well as user-defined types, need not be atomic. Aside from the library functions designed for that purpose, there is no guarantee of atomic read-modify-write, such as in the case of increment or decrement.
+
+## §ref-span-safety Reference variables and returns
+
+### §ref-span-safety-general General
+
+A *reference variable* is a variable that refers to another variable, called the *referent*. Therefore, a *reference variable* does not store its value. Instead, a *reference variable* refers to its *referent*. Thus, the value of a reference variable is always the same as its referent variable. A *reference variable* is a local variable declared with the `ref` modifier, or an instance of a `ref struct` type.
```

Comment:
I'm not sure what's meant by "or an instance of a `ref struct` type" here. Is an instance of a `ref struct` type always a variable? Is the point that if we call a method that returns a `ref struct`, we can treat that return value as a reference variable? An example would at least help me understand this - not sure yet whether it needs to be in the standard though!

Reply:
This is an area where the standard language gets much easier in C# 11. In C# 11, a `ref struct` can contain `ref` fields. Therefore, returning a `ref struct` by value returns the `ref` field, which is a reference.

In C# 7.n (up to C# 11), types such as `Span<T>` have *ref-like* fields. They are effectively references, but the compiler doesn't allow it yet, so it's a special case.

Reply:
I added three examples in this section: one for a local `ref` variable. One for an instance of a `ref struct`, and one for a `return ref`. I think the `ref struct` sample should address the "fudginess".